File: BE/src/main/java/com/ssafy/e106/domain/subscription/dto/response/SubscriptionRecentUsageResponse.java
package com.ssafy.e106.domain.subscription.dto.response;

import com.ssafy.e106.domain.subscription.entity.CheckinRecord;

public record SubscriptionRecentUsageResponse(
    String cycleYm,
    String response,
    Long serviceId,
    String serviceName) {

  public static SubscriptionRecentUsageResponse from(CheckinRecord checkinRecord) {
    return new SubscriptionRecentUsageResponse(
        checkinRecord.getCycleYm(),
        checkinRecord.getResponse().name(),
        checkinRecord.getService().getServiceId(),
        checkinRecord.getService().getName());
  }
}

File: FE/app/src/main/java/com/ssafy/e106/data/dto/subscription/SubscriptionRecentUsageResponse.kt
package com.ssafy.e106.data.dto.subscription

import kotlinx.serialization.Serializable

@Serializable
data class SubscriptionRecentUsageResponse(
    val cycleYm: String,
    val response: String,
    val serviceId: Long? = null,
    val serviceName: String? = null,
)
